### ISSUE TITLE
Cubemap Video

### ISSUE THREAD

**Issue by ruiluis** [mpetroff/pannellum#435]
is there support for a cubemap video instead off equirectangular video?

**Comment by mpetroff** [comment]
No, there isn't. That would require synchronizing six different videos.

**Comment by arpu** [comment]
hello why 6 different videos?  its ony one video 
this is one frame https://learnopengl.com/img/advanced/cubemaps_skybox.png

**Comment by mpetroff** [comment]
Pannellum handles cubemaps as six separate images, since WebGL (and OpenGL) handles cubemaps as six separate images. The example you provide is just a visualization of the format; the images are not normally stored that way since it would be very inefficient due to the unused space. Quoting from [the page](https://learnopengl.com/#!Advanced-OpenGL/Cubemaps) that image is from, "some resources provide the skyboxes in a [combined format] in which case you'd have to manually extract the 6 face images, but in most cases they're provided as 6 single texture images."

**Comment by arpu** [comment]
yes you are right from my sample but facebook i think uses all 6 face images in one frame 
(https://github.com/facebook/transform360)  i  will do some test with this 
here some more information on facebook cubemap (tey use it for live 360 degree videos) 
https://code.facebook.com/posts/1638767863078802/under-the-hood-building-360-video/

**Comment by mpetroff** [comment]
I took a look, and Facebook uses a 3x2 grid layout, so there isn't any wasted space. It's an interesting concept, as it has a more uniform distribution of pixels, although still not optimal (Facebook also has a newer ["barrel" projection](https://code.facebook.com/posts/118926451990297/enhancing-high-resolution-360-streaming-with-view-prediction/) which further increases the uniformity of the pixel distribution). For a while, I've been meaning to experiment with using the [COBE sky cube](http://tdc-www.harvard.edu/wcstools/projections/csc.html) projection to improve the efficiency of Pannellum's multiresolution images (which have separate images for each cube face). It might be interesting to try to combine it with Facebook's grid method for use with videos.

**Comment by ruiluis** [comment]
i am also working in this, as soon i have a solution i will posted.
just another info. facebook has now pass the cube ideia and is now using
triangles because of the bandwidth (
https://code.facebook.com/posts/1126354007399553/next-generation-video-encoding-techniques-for-360-video-and-vr/)
..
but for starting the cube format should be a good improvement

------
Give a man a fish and you feed him for a day. Teach him how to fish and you
feed him for a lifetime.

**Comment by mpetroff** [comment]
Actually, they already stopped using the pyramid format too. See https://github.com/facebook/transform360/issues/7#issuecomment-275894293.
